api/client/history.go

```diff
@@ -40,38 +41,40 @@ func (cli *DockerCli) CmdHistory(args ...string) error {
 	}
 
 	w := tabwriter.NewWriter(cli.out, 20, 1, 3, ' ', 0)
-	if !*quiet {
-		fmt.Fprintln(w, "IMAGE\tCREATED\tCREATED BY\tSIZE\tCOMMENT")
-	}
-
-	for _, entry := range history {
-		if *noTrunc {
-			fmt.Fprintf(w, entry.ID)
-		} else {
-			fmt.Fprintf(w, stringid.TruncateID(entry.ID))
-		}
-		if !*quiet {
-			if *human {
-				fmt.Fprintf(w, "\t%s ago\t", units.HumanDuration(time.Now().UTC().Sub(time.Unix(entry.Created, 0))))
-			} else {
-				fmt.Fprintf(w, "\t%s\t", time.Unix(entry.Created, 0).Format(time.RFC3339))
-			}
 
+	if *quiet {
+		for _, entry := range history {
 			if *noTrunc {
-				fmt.Fprintf(w, "%s\t", strings.Replace(entry.CreatedBy, "\t", " ", -1))
+				fmt.Fprintf(w, "%s\n", entry.ID)
 			} else {
-				fmt.Fprintf(w, "%s\t", stringutils.Truncate(strings.Replace(entry.CreatedBy, "\t", " ", -1), 45))
+				fmt.Fprintf(w, "%s\n", stringid.TruncateID(entry.ID))
 			}
+		}
+		w.Flush()
+		return nil
+	}
 
-			if *human {
-				fmt.Fprintf(w, "%s\t", units.HumanSize(float64(entry.Size)))
-			} else {
-				fmt.Fprintf(w, "%d\t", entry.Size)
-			}
+	fmt.Fprintln(w, "IMAGE\tCREATED\tCREATED BY\tSIZE\tCOMMENT")
+	for _, entry := range history {
+		imageID := stringid.TruncateID(entry.ID)
+		createdBy := stringutils.Truncate(strings.Replace(entry.CreatedBy, "\t", " ", -1), 45)
+		if *noTrunc {
```

Comment:
This feels a little odd to assume the longer path (meaning trunc everything) and then undo it. Seems it would be easier, and more optimal, to do the minimum first, then trunc only when needed:
```
imageID := entry.ID
createdBy := strings.Replace(entry.CreatedBy, "\t", " ", -1)
if *noTrunc == false {
  createdBy = stringutils.Truncate(createdBy)
  imageID = stringid.TruncateID(entry.ID)
}
```